index/secondary_index_manager.cc

```diff
@@ -374,4 +374,16 @@ std::optional<std::function<std::unique_ptr<custom_index>()>> secondary_index_ma
     }
 }
 
+std::optional<std::unique_ptr<custom_index>> secondary_index_manager::get_custom_class(const index_metadata& im) {
+    auto it = im.options().find("class_name");
```

Comment:
We already have a name for this "class_name" - 
db::index::secondary_index::custom_index_option_name
and also cql3::statements::index_target::custom_index_option_name
I think you should use one of them - and perhaps think why we have two :-(